SteamWorkshopService/SteamWorkshopService.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            //if (!IsReady)
+            //    throw new Exception("SteamWorkshopService not initialized!");
+
+            using (dynamic remoteStorage = SteamKit2.WebAPI.GetInterface("ISteamRemoteStorage"))
+            {
+                KeyValue allFilesDetails = null ;
+                remoteStorage.Timeout = TimeSpan.FromSeconds(30);
+                allFilesDetails = await Task.Run(delegate {
+                    try
+                    {
+                        return remoteStorage.GetPublishedFileDetails1(
+                            itemcount: workshopIds.Count(), 
+                            publishedfileids: workshopIds, 
+                            method: HttpMethod.Post);
+                    }
+                    catch (HttpRequestException e)
+                    {
+                        Log.Error($"Fetching File Details failed: {e.Message}");
+                        return null;
+                    }
+                });
+                if (allFilesDetails == null)
+                    return null;
+                //fileDetails = remoteStorage.Call(HttpMethod.Post, "GetPublishedFileDetails", 1, new Dictionary<string, string>() { { "itemcount", workshopIds.Count().ToString() }, { "publishedfileids", workshopIds.ToString() } });
+                var detailsList = allFilesDetails?.Children.Find((KeyValue kv) => kv.Name == "publishedfiledetails")?.Children;
+                var resultCount = allFilesDetails?.GetValueOrDefault<int>("resultcount");
+                if( detailsList == null || resultCount == null)
+                {
+                    Log.Error("Received invalid data: ");
+#if DEBUG
+                    if(allFilesDetails != null)
+                        PrintKeyValue(allFilesDetails);
+                    return null;
+#endif
+                }
+                if ( detailsList.Count != workshopIds.Count() || resultCount != workshopIds.Count())
+                {
+                    Log.Error($"Received unexpected number of fileDetails. Expected: {workshopIds.Count()}, Received: {resultCount}");
+                    return null;
+                }
+
+                var result = new Dictionary<ulong, PublishedItemDetails>();
+                for( int i = 0; i < resultCount; i++ )
+                {
+                    var fileDetails = detailsList[i];
+
+                    var tagContainer = fileDetails.Children.Find(item => item.Name == "tags");
+                    List<string> tags = new List<string>();
+                    if (tagContainer != null)
+                        foreach (var tagKv in tagContainer.Children)
+                        {
+                            var tag = tagKv.Children.Find(item => item.Name == "tag")?.Value;
+                            if( tag != null)
+                                tags.Add(tag);
+                        }
+
+                    result[workshopIds.ElementAt(i)] = new PublishedItemDetails()
```

Comment:
`IEnumerable.ElementAt(i)` should never be used since it has to iterate the enumerable each time you call it.  Additionally if for some reason the order of the returned results varies this will fail.  Prefer keying using the `publishedfileid`.

Reply:
Can't use `publishedFileId` in that case, but will try to get rid of `ElementAt(i)` :+1: